updater.go

```diff
@@ -50,8 +56,159 @@ func NewUpdater(source UpdateSource, config Config, log logging.Logger) *Updater
 	}
 }
 
-// Update performs an update
+// Update checks, downloads and performs an update
 func (u *Updater) Update(ctx Context) (*Update, error) {
-	// TODO
-	return nil, nil
+	options := ctx.UpdateOptions()
+	update, err := u.checkForUpdate(ctx, options)
+	if err != nil {
+		return nil, err
+	}
+	if update == nil {
+		// No update available
+		return nil, nil
+	}
```

Comment:
I kind of like being really verbose here and following the if err != nil pattern and then saying hey, no error but we don't have an update and that's ok...

If this seems crazy though I can change it to your suggestion.